lib-injection/auto_inject.rb

```diff
@@ -0,0 +1,56 @@
+if !ENV['skip_autoinject']
+  if system 'skip_autoinject=true bundle show ddtrace'
+    puts 'ddtrace already installed... skipping auto-injection'
+  else
+    version = "<DD_TRACE_VERSION_TO_BE_REPLACED>"
+    sha = "<DD_TRACE_SHA_TO_BE_REPLACED>"
+
+    condition =
+      if !version.empty?
+        # For public release
+        "--version '#{version.gsub(/^v/, '')}'"
+      elsif !sha.empty?
+        # For internal testing
+        "--github 'datadog/dd-trace-rb' --ref '#{sha}'"
+      else
+        puts "NO VERSION"
+      end
+
+    puts "ddtrace is not installed... Perform auto-injection... for dd-trace-rb"
+
+    require 'bundler'
+    require 'fileutils'
+
+    gemfile  = Bundler::SharedHelpers.default_gemfile
+    lockfile = Bundler::SharedHelpers.default_lockfile
+
+    datadog_gemfile = gemfile.dirname + ("datadog-#{gemfile.basename}")
+    datadog_lockfile = lockfile.dirname + ("datadog-#{lockfile.basename}")
+
+    # Copies for trial
+    FileUtils.cp gemfile, datadog_gemfile
+    FileUtils.cp lockfile, datadog_lockfile
+
+    if system "skip_autoinject=true BUNDLE_GEMFILE=#{datadog_gemfile} bundle add ddtrace #{condition} --require ddtrace/auto_instrument"
```

Comment:
I recommend checking for usual cases where this command will fail, like when the bundle is in `deployment` mode or `frozen` (using the return value from `Bundler.frozen_bundle?`):
https://github.com/rubygems/rubygems/blob/451ac56078a3deaacef640906070d3d8f7502570/bundler/lib/bundler.rb#L220-L224